bot/dialog-manager/core/src/test/kotlin/ai/tock/bot/processor/TickStoryProcessorTargetStoryTest.kt

```diff
@@ -0,0 +1,214 @@
+                    contextNames = setOf(contextJeVeuxJouer.name)
+                )
+            )
+        ),
+        TickIntent(
+            intentName = intentNon,
+            associations = setOf(
+                TickIntentAssociation(
+                    actionName = actionTicTacToe.name,
+                    contextNames = setOf(contextJeNeVeuxPasJouer.name)
+                )
+            )
+        ),
+    )
+
+    private val tickConfigBonjourRobot = TickConfiguration(
+        stateMachine = stateMachine,
+        contexts = setOf(contextJeVeuxJouer, contextJeNeVeuxPasJouer, contextDev1),
+        actions = setOf(actionBonjour, actionVeuxTuJouer, actionTicTacToe, actionTantPis, actionTantMieux, actionAurevoir),
+        intentsContexts = intentsContexts,
+        unknownHandleConfiguration = TickUnknownConfiguration(),
+        storySettings = TickStorySettings.default
+    )
+
+    private val tickSender = mockk<TickSender>()
+
+    @Test
+    fun `process when user intent is "non" then sender send actionTantMieuxAnswerId and processor redirect to target story`() {
+
+        val sendAnswersCaptured = mutableListOf<String>()
+        val endAnswersCaptured = mutableListOf<String>()
+
+        val produceProcessor: TSupplier<TickStoryProcessor> = {
+            TickStoryProcessor(
+                session = TickSession(
+                    currentState = stateTicTacToe.id,
+                    contexts = setOf(contextDev1, contextDev2).associate { it.name to null },
+                    ranHandlers=listOf(actionBonjour.name, actionVeuxTuJouer.name, actionTicTacToe.name),
+                    objectivesStack = listOf(stateAurevoir.id),
+                    handlingStep = TickActionHandlingStep(repeated=1, action=actionTicTacToe.name)),
+                configuration = tickConfigBonjourRobot,
+                sender = tickSender,
+                endingStoryRuleExists = false
+            )
+        }
+
+        val mockBehaviours: TRunnable = {
+            every { tickSender.sendById(capture(sendAnswersCaptured)) } answers {}
+            every { tickSender.endById(capture(endAnswersCaptured)) } answers {}
+        }
+
+        val processCall: TFunction<TickStoryProcessor?, ProcessingResult> = {
+            it!!.process(TickUserAction(intentNon))
+        }
+
+        val checkResult: TConsumer<ProcessingResult?> = {
+            assertNotNull(it)
+            assert(it is Redirect)
+
+            assertEquals(actionTantPis.targetStory, (it as Redirect).storyId)
+
+            assertEquals(1, sendAnswersCaptured.size)
+            assertEquals(0, endAnswersCaptured.size)
+
+            verify(exactly = 1) { tickSender.sendById(actionTantPis.answerId!!) }
+            verify(exactly = 0) { tickSender.endById(any()) }
+            verify(exactly = 0) { tickSender.end() }
+        }
+
+        TestCase<TickStoryProcessor, ProcessingResult>("process a tick user action - cas 1")
```

Comment:
```suggestion
        TestCase<TickStoryProcessor, ProcessingResult>("process a tick user action")
```
Pas besoin de dire cas 1 je pense